src/catalog/trigger_catalog.cpp

```diff
@@ -0,0 +1,322 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// query_metrics_catalog.cpp
+//
+// Identification: src/catalog/query_metrics_catalog.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/catalog.h"
+#include "catalog/trigger_catalog.h"
+#include "catalog/catalog_defaults.h"
+#include "executor/logical_tile.h"
+#include "common/macros.h"
+#include "common/logger.h"
+
+namespace peloton {
+namespace catalog {
+
+TriggerCatalog *TriggerCatalog::GetInstance(
+    concurrency::Transaction *txn) {
+  static std::unique_ptr<TriggerCatalog> trigger_catalog(
+      new TriggerCatalog(txn));
+
+  return trigger_catalog.get();
+}
+
+TriggerCatalog::TriggerCatalog(concurrency::Transaction *txn)
+    : AbstractCatalog("CREATE TABLE " CATALOG_DATABASE_NAME
+                      "." TRIGGER_CATALOG_NAME
+                      " ("
+                      "trigger_oid           INT NOT NULL, "
```

Comment:
The names of these columns do not match "pg_trigger"

https://www.postgresql.org/docs/current/static/catalog-pg-trigger.html